examples/distributed/elementwise.py

```diff
@@ -0,0 +1,45 @@
+import numpy
+import cupy
+from cupy.cuda import Device, Stream
+from cupyx.distributed.array import distributed_array
+
+
+SIZE = 300 * 300
+
+SHAPE = (300, 300)
+
+INDEX_MAP_A = {
+    0: (slice(200)),  # arr[:200]
+    1: (slice(200, None)),  # arr[200:]
+}
+
+INDEX_MAP_B = {
+    0: (slice(None), slice(None, None, 3)),  # arr[:,  ::2]
+    1: (slice(None), slice(1, None, 2)),  # arr[:, 1::2]
+}
```

Comment:
It'd be nice to comment why we need these index maps, currently this sample assumes that users have the concept of sharding, which may or may not be the case.